coordinator/gscoordinator/utils.py

```diff
@@ -227,6 +228,39 @@ def get_graph_sha256(attr):
     return hashlib.sha256(graph_class.encode("utf-8")).hexdigest()
 
 
+def check_java_app_graph_consistency(
+    app_class, cpp_graph_type, java_class_template_str
+):
+    splited = cpp_graph_type.split("<")
+    java_app_type_params = java_class_template_str[:-1].split("<")[-1].split(",")
+    num_type_params = 0
+    if len(splited) != 2:
+        raise Exception("Unrecoginizable graph template str: {}".format(cpp_graph_type))
+    if splited[0] == "vineyard::ArrowFragment":
+        if app_class.find("Property") == -1:
+            logger.error("Expected property app")
+            return False
```

Comment:
I see no reason why you don't add the error message into the exception message (as you throw exception if `check_java_app_graph_consistency ` fails)...